### ISSUE TITLE
mdmonitor.service fail

### ISSUE THREAD

**Issue by jodumont** [coreos/bugs#1548]
# Issue Report #

## Bug ##
Failed Units: 1
  mdmonitor.service

### CoreOS Version ###
VERSION=1068.10.0
VERSION_ID=1068.10.0
BUILD_ID=2016-08-23-0220
PRETTY_NAME="CoreOS 1068.10.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"

### Environment ###
Generic BareMetal
CoreOS use as hypervisor

### Expected Behavior ###
run
### Actual Behavior ###
fail
### Reproduction Steps ###

  1. create a raid with mdadm -C ... -l1 -n2 /dev/sd[x|y]
  2. reboot

### Other Information ###
cat /usr/lib64/systemd/system/mdmonitor.service 
```
...
[Unit]
Description=MD array monitor
DefaultDependencies=no

[Service]
Environment=  MDADM_MONITOR_ARGS=--scan
EnvironmentFile=-/run/sysconfig/mdadm
ExecStartPre=-/usr/lib/systemd/scripts/mdadm_env.sh
ExecStart=/sbin/mdadm --monitor $MDADM_MONITOR_ARGS
...
```
### Could be fix by ###
Environment=  MDADM_MONITOR_ARGS=--scan **-m my@email.tld**

**Comment by r0p0s3c** [comment]
This seems to work, but testing brings up a more fundamental issue:

```
$ sudo /sbin/mdadm --monitor --scan --test -m me@me.tld
sh: /usr/sbin/sendmail: No such file or directory

```
Any ideas how to solve that?

**Comment by mischief** [comment]
Sounds like we should simply disable that service, since we will likely not
ship sendmail.

**Comment by mischief** [comment]
@r0p0s3c we've decided to go with a default of the syslog mode of mdadm --monitor.

can you try this dropin?


`/etc/systemd/system/mdmonitor.service.d/10-syslog.conf`:
```
[Service]
Environment=MDADM_MONITOR_ARGS="--scan --syslog"
```

if that works out, we'll make this the default for the future so that it works out of the box, and if anyone wants something different they can override it.

**Comment by r0p0s3c** [comment]
That looks like a saner default. I've added the dropin (though with:

`Environment="MDADM_MONITOR_ARGS=--scan --syslog"`

not sure why just quoting the args has an issue on my end) and the service is able to start without issue.

**Comment by mischief** [comment]
@r0p0s3c apologies, i believe i bungled the syntax of https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Environment=.

i'll prepare this change for the next alpha.

**Comment by mischief** [comment]
fixed by coreos/init#220
